app/component/StreetModeSelectorButton.js

```diff
@@ -61,35 +55,19 @@ export const StreetModeSelectorButton = (
   }
 
   let secondaryIcon;
-  let metroColor;
+  let secondaryColor;
 
-  if (name === 'bikeAndVehicle') {
-    const publicModes = plan.itineraries[0].legs.filter(
-      obj => obj.mode !== 'WALK' && obj.mode !== 'BICYCLE',
+  if (name === 'parkAndRide' || name === 'bikeAndVehicle') {
+    const transitItinerary = plan.itineraries.find(i =>
+      i.legs.find(l => l.transitLeg),
     );
-    if (publicModes.length > 0) {
-      const firstMode = getMode(publicModes[0], config);
-      secondaryIcon = `icon-icon_${firstMode}`;
-      if (firstMode === 'subway') {
-        metroColor = '#CA4000';
-      }
-    }
-  } else if (name === 'parkAndRide') {
-    let mode = 'rail';
-    for (let i = 0; i < plan.itineraries.length; i++) {
-      const publicModes = plan.itineraries[i].legs.filter(
-        obj =>
-          obj.mode !== 'WALK' && obj.mode !== 'BICYCLE' && obj.mode !== 'CAR',
-      );
-      if (publicModes.length > 0) {
-        mode = getMode(publicModes[0], config);
-        break;
-      }
-    }
+    const mode =
+      getExtendedMode(
+        transitItinerary?.legs.find(l => l.transitLeg),
+        config,
+      ) || 'rail';
     secondaryIcon = `icon-icon_${mode}`;
-    if (mode === 'subway') {
-      metroColor = '#CA4000';
-    }
+    secondaryColor = mode === 'subway' ? '#CA4000' : '';
```

Comment:
Hardcoded subway color feels a bit wrong.

Reply:
Not  only a bit. Fixed.